exporter/kinesisexporter/producer.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kinesisexporter
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	omnition "github.com/signalfx/omnition-kinesis-producer"
+	"github.com/signalfx/omnition-kinesis-producer/loggers/kpzap"
+	"go.uber.org/zap"
+)
+
+// producer provides the interface for a kinesis producer. The producer
+// implementation abstracts the interaction with the kinesis producer library
+// used from the exporter
+type producer interface {
```

Comment:
Is this an interface just to make testing easier or is there more to it?

Reply:
Yeah and also just makes it a bit easier to switch out the producer library if we ever decide to in the future